### PR TITLE
DS: I18N settings, create modal

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

Moves the I18N settings modal for creating a new locale to the new DS

### How to test it?

- Login the dashboard
- Select "Settings" on the main menu
- Select "I18N" in the sub menu
- Select the "Add a locale" button
- 
### Screenshot

The I18N settings modal opened

### CODE REVIEW

File: packages/core/helper-plugin/lib/src/components/Form/index.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-        errorNames.forEach(errorKey => {
-          const selector = `[name="${errorKey}"]`;
-
-          if (!errorEl) {
-            errorEl = document.querySelector(selector);
-          }
-        });
-
-        errorEl.focus();
+      const errorsInForm = formRef.current.querySelectorAll('[data-strapi-field-error]');
```
Comment (mfrachet) [inline packages/core/helper-plugin/lib/src/components/Form/index.js:10]: ℹ️ moving to the new data attribute in order to target elements like "buttons" in select

File: packages/core/helper-plugin/lib/src/components/Form/index.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
-import React, { useEffect } from 'react';
-import { Form, useFormikContext, getIn } from 'formik';
+import React, { useEffect, useRef } from 'react';
+import { Form, useFormikContext } from 'formik';
 
 const FormWithFocus = props => {
+  const formRef = useRef(null);
```
Comment (mfrachet) [inline packages/core/helper-plugin/lib/src/components/Form/index.js:5]: ℹ️ allows to query only inside the form, and not on the whole document

File: packages/plugins/i18n/admin/src/components/LocaleList/index.js [hunk 3]
```diff
@@ -48,27 +48,21 @@ const LocaleList = ({ canUpdateLocale, canDeleteLocale, onToggleCreateModal, isC
...
-              action={
-                onToggleCreateModal ? (
-                  <Button variant="secondary" startIcon={<AddIcon />} onClick={onToggleCreateModal}>
-                    {formatMessage({ id: getTrad('Settings.list.actions.add') })}
-                  </Button>
-                ) : null
-              }
-            />
-          </ContentLayout>
+          <EmptyStateLayout
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleList/index.js:51]: Here you can use the `CustomContentLayout` component instead:

```js
import { CustomContentLayout } from '@strapi/helper-plugin';

<CustomContentLayout
        shouldShowEmptyState={locales === undefined || locales?.length === 0}
        action={
          onToggleCreateModal ? (
            <Button variant="secondary" startIcon={<AddIcon />} onClick={onToggleCreateModal}>
              {formatMessage({ id: getTrad('Settings.list.actions.add') })}
            </Button>
          ) : null
        }
      >
        <LocaleTable
          locales={locales}
          onDeleteLocale={handleDeleteLocale}
          onEditLocale={handleEditLocale}
        />
      </CustomContentLayout>
```

File: packages/plugins/i18n/admin/src/components/LocaleList/index.js [hunk 4]
```diff
@@ -48,27 +48,21 @@ const LocaleList = ({ canUpdateLocale, canDeleteLocale, onToggleCreateModal, isC
...
+          />
         )}
       </ContentLayout>
 
-      <ModalCreate
-        isOpened={isCreating}
-        onClose={onToggleCreateModal}
-        alreadyUsedLocales={locales}
-      />
+      {isCreating && <ModalCreate onClose={onToggleCreateModal} />}
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleList/index.js:65]: I have an error when creating a locale:

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+      return !foundLocale;
+    });
+
+  const defaultOption = options[0];
+
+  return (
+    <Select
+      label={formatMessage({
+        id: getTrad('Settings.locales.modal.locales.label'),
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:35]: Can you add a defaultMessage?

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      })}
+      onClear={value ? onClear : undefined}
+      clearLabel={formatMessage({
+        id: getTrad('Settings.locales.modal.base.clearLabel'),
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:39]: Same here.

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      value={value || defaultOption}
+      onChange={selectedLocaleKey => {
+        const selectedLocale = options.find(locale => locale.label === selectedLocaleKey);
+
+        if (selectedLocale) {
+          onLocaleChange({ code: selectedLocale.label, displayName: selectedLocale.value });
+        }
+      }}
+    >
+      {isLoading
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:51]: Maybe you can show a loading indicator like in the users create modal ?

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        ? null
+        : options.map(option => (
+          <Option value={option.label} key={option.label}>
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:54]: I think you switch the label and the value:

```js
<Option value={option.value} key={option.value}>{option.label}</Option>
```

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+ * key in a formik form
+ */
+const LocaleSelect = React.memo(({ value, onLocaleChange, error, onClear }) => {
+  const { formatMessage } = useIntl();
+  const { defaultLocales, isLoading } = useDefaultLocales();
+  const { locales } = useLocales();
+
+  const options = (defaultLocales || [])
+    .map(locale => ({
+      label: locale.code,
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:21]:
```
label: locale.name,
value: locale.code
```

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      value: locale.name,
+    }))
+    .filter(({ label }) => {
+      const foundLocale = locales.find(({ code }) => code === label);
+
+      return !foundLocale;
+    });
+
+  const defaultOption = options[0];
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:30]:
```js
const defaultValue = value || options[0]?.value || '';
```

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    <Select
+      label={formatMessage({
+        id: getTrad('Settings.locales.modal.locales.label'),
+      })}
+      onClear={value ? onClear : undefined}
+      clearLabel={formatMessage({
+        id: getTrad('Settings.locales.modal.base.clearLabel'),
+      })}
+      error={error}
+      value={value || defaultOption}
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:42]:
```
value={defaultValue}
```

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+      onChange={selectedLocaleKey => {
+        const selectedLocale = options.find(locale => locale.label === selectedLocaleKey);
+
+        if (selectedLocale) {
+          onLocaleChange({ code: selectedLocale.label, displayName: selectedLocale.value });
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:47]: This needs to be switch to `{ code: selectedLocale.value, displayName: selectedLocale.label }`

File: packages/plugins/i18n/admin/src/components/ModalCreate/AdvancedForm.js [hunk 13]
```diff
@@ -1,44 +1,25 @@
...
-
-      <Text color="grey" fontSize="sm">
-        {formatMessage({
-          id: getTrad('Settings.locales.modal.advanced.setAsDefault.hint'),
-        })}
-      </Text>
-    </div>
+    <Checkbox
+      hint={formatMessage({
+        id: getTrad('Settings.locales.modal.advanced.setAsDefault.hint'),
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/ModalCreate/AdvancedForm.js:14]: Can you add a defaultMessage?

File: packages/plugins/i18n/admin/src/components/ModalCreate/AdvancedForm.js [hunk 14]
```diff
@@ -1,44 +1,25 @@
...
+      })}
+      onChange={() => setFieldValue('isDefault', !values.isDefault)}
+      value={values.isDefault}
+    >
+      {formatMessage({
+        id: getTrad('Settings.locales.modal.advanced.setAsDefault'),
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/ModalCreate/AdvancedForm.js:20]: Same here.

File: packages/plugins/i18n/admin/src/components/ModalCreate/BaseForm.js [hunk 15]
```diff
@@ -1,103 +1,71 @@
...
-        <Inputs
+      <GridItem col={6}>
+        <TextInput
+          name="displayName"
           label={formatMessage({
             id: getTrad('Settings.locales.modal.locales.displayName'),
           })}
-          name="displayName"
-          description={formatMessage({
+          hint={formatMessage({
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/ModalCreate/BaseForm.js:53]: Same here can you add the defaultMessages?

File: packages/plugins/i18n/admin/src/schemas.js [hunk 16]
```diff
@@ -1,7 +1,10 @@
 import { object, string } from 'yup';
 
 const localeFormSchema = object().shape({
-  displayName: string().max(50, 'Settings.locales.modal.locales.displayName.error'),
+  code: string().required(),
+  displayName: string()
+    .max(50, 'Settings.locales.modal.locales.displayName.error')
+    .required(),
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/schemas.js:7]:
```js
import { translatedErrors as errorsTrads } from '@strapi/helper-plugin';

.required(errorsTrads.required)
```

File: packages/plugins/i18n/admin/src/translations/en.json [hunk 17]
```diff
@@ -25,8 +25,9 @@
   "Settings.locales.modal.advanced.setAsDefault.hint": "One default locale is required, change it by selecting another one",
   "Settings.locales.modal.advanced.settings": "Settings",
   "Settings.locales.modal.base": "Base settings",
+  "Settings.locales.modal.base.clearLabel": "Clear label",
   "Settings.locales.modal.create.alreadyExist": "This locale already exists",
-  "Settings.locales.modal.create.confirmation": "Add locale",
+  "Settings.locales.modal.create.confirmation": "Save",
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/translations/en.json:30]: No need to recreate this one it already exists the id is : `app.components.Button.save`

File: packages/plugins/i18n/admin/src/translations/en.json [hunk 18]
```diff
@@ -25,8 +25,9 @@
   "Settings.locales.modal.advanced.setAsDefault.hint": "One default locale is required, change it by selecting another one",
   "Settings.locales.modal.advanced.settings": "Settings",
   "Settings.locales.modal.base": "Base settings",
+  "Settings.locales.modal.base.clearLabel": "Clear label",
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/translations/en.json:28]: Can you put this trads in the admin package so it is available globally?

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/helper-plugin/lib/src/components/Form/Form.stories.mdx [hunk 19]
```diff
@@ -0,0 +1,75 @@
+<!--- EmptyStateLayout.stories.mdx --->
+
+import { Meta, ArgsTable, Canvas, Story } from '@storybook/addon-docs';
```
Comment (soupette) [inline packages/core/helper-plugin/lib/src/components/Form/Form.stories.mdx:3]: 👍

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+      label: locale.name,
+      value: locale.code,
+    }))
+    .filter(({ value: v }) => {
+      const foundLocale = locales.find(({ code }) => code === v);
+
+      return !foundLocale;
+    });
+
+  const computedValue = value || '';
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:39]: `const computedValue = value || options[0]?.value || '';`

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/plugins/i18n/admin/src/components/LocaleSelect/index.js (hunk 20 above)
Comment (mfrachet) [reply packages/plugins/i18n/admin/src/components/LocaleSelect/index.js:39]: it won't match the default display name if we do that

**Pull request merged**